controllers/storageclient_controller.go

```diff
@@ -210,9 +215,47 @@ func (r *StorageClientReconciler) reconcilePhases() (ctrl.Result, error) {
 			if err := r.reconcileClusterResourceQuota(clusterResourceQuotaSpec); err != nil {
 				return reconcile.Result{}, err
 			}
+		case "Secret":
+			data := &corev1.Secret{}
+			if err := json.Unmarshal(eResource.Data, &data); err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to unmarshall secret: %v", err)
+			}
+			secret := &corev1.Secret{}
+			secret.Name = eResource.Name
+			secret.Namespace = r.storageClient.Namespace
+			_, err := controllerutil.CreateOrUpdate(r.ctx, r.Client, secret, func() error {
+				if err := r.own(secret); err != nil {
+					return err
+				}
+				secret.Data = data.Data
+				return nil
+			})
+			if err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to create or update secret %s :%v", secret.Name, err)
```

Comment:
1. You have an issue with the position of the char ":"
2. Please capture the namespaced name 
2. Consider splitting it into a multiline statement (As of the length of the current line) 

```suggestion
				return reconcile.Result{}, fmt.Errorf(
					"failed to create or update secret %v: %v", 
					client.ObjectKeyFromObject(secret) , 
					err,
				)
```

Reply:
Updated